integration/init_test.go

```diff
@@ -56,21 +59,55 @@ func TestInit(t *testing.T) {
 				"-a", `{"builder":"Docker","payload":{"path":"leeroy-web/Dockerfile"},"image":"gcr.io/k8s-skaffold/leeroy-web"}`,
 			},
 		},
+	}
+	for _, test := range tests {
+		testutil.Run(t, test.name, func(t *testutil.T) {
+			initArgs := append([]string{"--force"}, test.args...)
+			skaffold.Init(initArgs...).InDir(test.dir).WithConfig("skaffold.yaml.out").RunOrFail(t.T)
+
+			checkGeneratedConfig(t, test.dir)
+
+			skaffold.Diagnose().InDir(test.dir).WithConfig("skaffold.yaml.out").RunOrFail(t.T)
+		})
+	}
+}
+
+func TestInitCompose(t *testing.T) {
+	if testing.Short() || RunOnGCP() {
+		t.Skip("skipping kind integration test")
+	}
+
+	tests := []struct {
+		name string
+		dir  string
+		args []string
+	}{
 		{
 			name: "compose",
 			dir:  "testdata/init/compose",
 			args: []string{"--compose-file", "docker-compose.yaml"},
 		},
 	}
 	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			ns, _, deleteNs := SetupNamespace(t)
+		testutil.Run(t, test.name, func(t *testutil.T) {
+			ns, _, deleteNs := SetupNamespace(t.T)
 			defer deleteNs()
 
 			initArgs := append([]string{"--force"}, test.args...)
-			skaffold.Init(initArgs...).InDir(test.dir).WithConfig("skaffold.yaml.out").RunOrFail(t)
+			skaffold.Init(initArgs...).InDir(test.dir).WithConfig("skaffold.yaml.out").RunOrFail(t.T)
+
+			checkGeneratedConfig(t, test.dir)
 
-			skaffold.Run().InDir(test.dir).WithConfig("skaffold.yaml.out").InNs(ns.Name).RunOrFail(t)
+			skaffold.Run().InDir(test.dir).WithConfig("skaffold.yaml.out").InNs(ns.Name).RunOrFail(t.T)
```

Comment:
if you changed this to Diagnose above, we could do this here too, right?

Reply:
This sample also generates kubernetes yaml, so it's better to do a full run. I'll add a comment